### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by lekegitrepo**
### Enumerable Methods.
This project is about rewriting the ruby enumerable methods.

### CODE REVIEW

**Comment by lekegitrepo** [comment]
I enabled stickler and added the necessary files.

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
Hi @lekegitrepo your approach to the implementation of this is cool :clap: Please see the below things that is needed to be accomplished. You may need to refer to self more to fulfil some of those, kudos on your first attempt :+1:  Please request for a re-review when you are done :pray: 

my_each returns an Enumerator if no block is given
my_each_with_index returns an enumerator if no block is given
my_select returns an enumerator if no block is given
my_all? returns true if the block never returns false or nil
my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
my_none? when no block or argument is given returns true only if none of the collection members is true
my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
my_count returns the number of items in enum through enumeration
my_count counts the number of items in enum that are equal to item if an argument is given
my_map returns a new array with the results of running block once for every element in enum.
my_map returns an Enumerator if no block is given
my_inject combines all elements of enum by applying a binary operation, specified by a block
my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+# arr.my_any? { |i| puts i < 0 } # => false
+
+# puts arr.my_none?{|a| a.nil? } # => true
+
+# puts arr.my_count # => 5
+
+# puts arr.my_map { |i| i * 2} # => [2, 4, 6, 8, 10]
+
+# puts multiply_els([2, 4, 5])
```
Comment (bolah2009) [inline enumerable.rb]: #### `REQUIRED`

Consider deleting commented out codes. If you need to review a previous version of your code, use git history.

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      my_each { |i| return false if i =~ param }
+    elsif param.nil?
+      my_each { |i| return false if i }
+    else
+      my_each { |i| return false if i == param }
+    end
+    true
+  end
+
+  def my_count(items = nil)
```
Comment (bolah2009) [inline enumerable.rb:78]: #### REQUIRED

There is a :bug: with `my_count` method. Please, let's fix this.

```ruby
$ irb
irb(main):001:0> require_relative 'enumerable'
=> true
irb(main):002:0> [1,2,3,4,5].count(3)
=> 1
irb(main):003:0> [1,2,3,4,5].my_count(3)
=> 0
```

**Review by bolah2009 — CHANGES_REQUESTED** [review]
## Nice job @lekegitrepo  :smile: 

#### Awesome progress... :muscle: All methods are working just fine with only one exception. :clap: 

Please see comment below, fix it and submit another code review request. 

- Update PR Description
   update-PR-description

#### :recycle:  Make these changes and submit another code review request. :+1: 
#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Review by bolah2009 — APPROVED** [review]
## Awesome :clap: 

Good job making all suggested changes. :100: 


### It's a :+1: from me.


#### :heavy_check_mark:  This project is approved.  You can merge to master. :+1: 

As suggested in the previous review, please edit and update this PR description (:point_down:)

lekepr

**Pull request merged**